### DISCUSSION TITLE
김세훈 스토리

### DISCUSSION THREAD

**Discussion by ResinCRT** [ondal-house/zu-story#26] [스토리 작성 안건]
# 프로젝트 개요
술이술술 (술 정보가 술술 나온다는 뜻으로...?)

술에 관한 커뮤니티, 술에 대한 정보를 얻을 수도 있고 익명의 리뷰를 통해 신뢰도 측정 가능

# 기능
세부적인 기능은 다 너무 잘써놔서 할게읍네

- 가격비교
    해외 정가랑 국내 소매가가 제멋대로인데 다나와처럼 이에 대한 정보를 알 수 있으면 좋을지도? 

# 기술 고려사항

## FE
- 웹? 앱?
      - 앱 -> 맥 & i없찐, 반대로 안드 없으면 실기기테스트가 어려움
      - 웹 -> 먹던거 또 먹기 싫을 수도 있음
       저는 문제점 해결이 안되면 웹 해야 되지 않을까 생각이 듬

BE쪽은 잘 몰라서 ㅎㅎ

## DB
- 프리티어용 db.t2.micro는 db 띄우는것도 힘들어할 것 같긴한데 테스트해보고 노답이면 바꾸면 될 듯
- 그 경우 RDB는 mysql용 aurora serverless 쓰는게 젤 적게 먹힐 것 같고 그게 아니면 DynamoDB정도? 
- Aurora RDB는 mysql / postgresql을 지원하는데 DB쪽 기능을 심화해서 건들 일이 없을 것 같으니 편한 대로 고르면 될 듯함

## 서버
- EC2 : 프리티어가 micro인스턴스를 지원하긴 함, 근데 micro는 성능이 좀 안 좋음
- ECS : 프리티어용 ec2 micro 띄워놓고 배포하는건 공짜인 것 같음
- EKS : 기본비용이 꽤 비쌈, 프리티어 제공X, 대신 각종 솔루션을 조합해서 내부 네트워킹 / HA / 오토스케일링 / 보안 다 해결 가능. 러닝커브는 매우 높지만 저랑 기용이가 찍먹은 시켜드릴 수 있음, 클라우드 정수를 느끼고 싶으면 추천
- Lambda : 프리티어도 지원, 사실상 공짜로 쓸 수 있음, 유지보수 어려움

## SSL 관련
 - ALB가 총합해서 월별 750시간 공짜니까 하나 정도는 풀로 돌릴 수 있을 듯? (도메인 붙여서 https 통신하는거)

## CI/CD
- 구성을 최대한 단순하게 할거면 github action 붙이는게 제일 쉬워보임(github 연동 부분 아예 생략 가능)
- Saas 형 러너를 제공하는데 사양이 생각보다 괜찮고(실성능은 테스트해봐야겠지만) public repo에 한해 2000분 공짜라 그냥 이거써도 될듯(러너 구성 필요 X)
- 이 경우 파이프라인 스크립트만 쓰면 되니 공수 매우 줄어들 것 같음

## 테스트
- 소나큐브나 junit 등등 적용한다치면 로컬에서만 테스트 아님 테스트 자동화?
- 자동화시 CI/CD에 껴서 할 건지, 아님 걍 로컬 테스트용 도커 환경만 compose같은걸로 만들어서 공유할건지, 로컬 shell 스크립트에서만 할건지
- 테스트결과 MR화면에 포함해서 결과 확인 같은 것도 할 수 있음
- MR/테스트 환경 구축은 필요한지? 
- 어찌됫든 테스트 수행할꺼면 적용방식과 범위가 궁금함
- CICD 구성 공수가 줄어들면 테스트 자동화 쪽으로 집중해봐도 되지 않을까요?

#### 그 외 의사결정 필요한거
보안이슈, 유저풀 관리, 비용 정산, 인프라 운영(Ops) 책임?

**Comment by ypd01018** [comment]
테스트자동화 넣죠 ㅋㅋㅋ
테스트자동화 없는 ci/cd 는 ci/cd라 부르면 안된다고 배웠습니다 ㅋㅋ

**Comment by gychoi0916** [comment]
저는... 찍먹 못시켜드려요~ 저도 찍먹하러왔습니다~ ㅎ.ㅎ ㅋㅋㅋㅋ
